src/plugins/condition/ConditionManager.js

```diff
@@ -0,0 +1,254 @@
+                    criteria: isDefault ? [] : [{
+                        telemetry: '',
+                        operation: '',
+                        input: '',
+                        metadata: ''
+                    }]
+                },
+                summary: ''
+            };
+        }
+        let conditionDomainObjectKeyString = this.openmct.objects.makeKeyString(conditionObj.identifier);
+        let newDomainObject = this.instantiate(conditionObj, conditionDomainObjectKeyString);
+
+        return newDomainObject.useCapability('adapter');
+    }
+
+    addCondition(isDefault, index) {
+        this.createAndSaveConditionDomainObject(!!isDefault, index);
+    }
+
+    cloneCondition(conditionConfigurationId, index) {
+        this.openmct.objects.get(conditionConfigurationId).then((conditionConfiguration) => {
+            this.createAndSaveConditionDomainObject(false, index, conditionConfiguration);
+        });
+    }
+
+    createAndSaveConditionDomainObject(isDefault, index, conditionConfiguration) {
+        let newConditionDomainObject = this.createConditionDomainObject(isDefault, conditionConfiguration);
+        //persist the condition domain object so that we can do an openmct.objects.get on it and only persist the identifier in the conditionCollection of conditionSet
+        this.openmct.objects.mutate(newConditionDomainObject, 'created', new Date());
+        if (index !== undefined) {
+            this.domainObject.configuration.conditionCollection.splice(index + 1, 0, newConditionDomainObject.identifier);
+        } else {
+            this.domainObject.configuration.conditionCollection.unshift(newConditionDomainObject.identifier);
+        }
+        this.persist();
+    }
+
+    removeCondition(identifier) {
+        let found = this.findConditionById(identifier);
+        if (found) {
+            let index = found.index;
+            let condition = this.conditionCollection[index];
+            let conditionIdAsString = condition.id;
+            condition.destroyCriteria();
+            condition.off('conditionResultUpdated', this.handleConditionResult.bind(this));
+            this.conditionCollection.splice(index, 1);
+            this.domainObject.configuration.conditionCollection.splice(index, 1);
+            if (this.conditionResults[conditionIdAsString] !== undefined) {
+                delete this.conditionResults[conditionIdAsString];
+            }
+            this.persist();
+            this.handleConditionResult();
+        }
+    }
+
+    findConditionById(identifier) {
+        let found;
+        for (let i=0, ii=this.conditionCollection.length; i < ii; i++) {
+            if (this.conditionCollection[i].id === this.openmct.objects.makeKeyString(identifier)) {
+                found = {
+                    item: this.conditionCollection[i],
+                    index: i
+                }
+                break;
+            }
+        }
+
+        return found;
+    }
```

Comment:
You can eliminate defining `found` and defining `ii`
```suggestion
    findConditionById(identifier) {
        for (let i=0, i < this.conditionCollection.length; i++) {
            if (this.conditionCollection[i].id === this.openmct.objects.makeKeyString(identifier)) {
                return {
                    item: this.conditionCollection[i],
                    index: i
                }
            }
        }
    }
```

Reply:
With this refactored approach, if conditionCollection is an empty array, findConditionById would not return anything.

Reply:
in both versions the return value would be undefined